File: java/0228_Summary_Ranges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        int n = nums.length, i = 0;
        List<String> ans = new ArrayList<>();

        while (i < n) {
            int cur = nums[i];
            while (i + 1 < n && nums[i + 1] - nums[i] == 1) {
                i++;
            }

            if (nums[i] != cur) {
                ans.add(cur + "->" + nums[i]);
            } else {
                ans.add(String.valueOf(cur));
            }

            i++;
        }

        return ans;
    }
}

File: python/0228_Summary_Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)
        ans = []
        i = 0

        while i < n:
            cur = nums[i]
            while i + 1 < n and nums[i + 1] - nums[i] == 1:
                i += 1
                
            if nums[i] == cur:
                ans.append(str(cur))
            else:
                ans.append("{}->{}".format(cur, nums[i]))
            i += 1

        return ans